kmod/core/core.c

```diff
@@ -839,14 +892,25 @@ static int kpatch_module_notify(struct notifier_block *nb, unsigned long action,
 
 	pr_notice("patching newly loaded module '%s'\n", object->name);
 
-	/* run any user-defined load hooks */
-	list_for_each_entry(hook, &object->hooks_load, list)
-		(*hook->hook)();
+	/* run any user-defined pre-patch callbacks */
+	if (object->pre_patch_callback) {
+		object->callbacks_enabled = true;
+		ret = (*object->pre_patch_callback)(object);
+		if (ret) {
+			object->callbacks_enabled = false;
+			pr_err("pre-patch callback failed!\n");
+			goto out;	/* and WARN */
```

Comment:
This stuff hurts my brain, but I think callbacks_enabled should always be true, unless there was an error with the pre_patch callback, right?  So shouldn't callbacks_enabled be set unconditionally before checking for object->pre_patch_callback?